deeprank_gnn/models/structure.py

```diff
@@ -133,23 +138,23 @@ def get_pssm(self):
         return pssm[self]
 
     @property
-    def number(self):
+    def number(self) -> int:
         return self._number
 
     @property
     def chain(self):
         return self._chain
 
     @property
-    def amino_acid(self):
+    def amino_acid(self) -> AminoAcid:
         return self._amino_acid
 
     @property
     def atoms(self):
```

Comment:
What about the return type here?

Reply:
That's a good one 😉  It should be `List[Atom]`, but we haven't defined `Atom` here yet (we do that below in the `structure` module). The proper way to do this would be to work with interfaces (i.e. an `AtomInterface` that we define in a separate module), but I didn't go for it because it will increase the complexity of the code and I am not sure if we will have much benefits from it. Another way would be to reorder the class definitions in `models/structure.py` so that `Atom` is defined before `Residue`. But then we have the same problem for `Residue.chain`.